### ISSUE TITLE
Doesn't work with legacy devices[issue]: 

### ISSUE THREAD

**Issue by Gravarty** [ventoy/Ventoy#2418]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0,91

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### Try alternative boot mode

Yes. I have tried them, but the bug still exist.

### BIOS Mode

Legacy BIOS Mode

### Partition Style

MBR

### Disk Capacity

128

### Disk Manufacturer

Kingston

### Image file checksum (if applicable)

Yes.

### What happened?

It's not possible to install Windows 10 booted from a Ventoy USB drive.
The error is "We couldn't create a new partition or locate an existing one. For more information, see the Setup log files".
Manually formating the drive using diskpart and copying over the windows installation files with DISM also corrupts the Ventoy USB partition (even though i never formated or even touched the USB partition is diskpart or setup).

**Comment by steve6375** [comment]
1. Boot to Ventoy in legacy BIOS MODE
2. Select Windows 10 ISO
3. Boot to Windows Setup
4. At Select Source Disk - DELETE ALL PARTIONS ON TARGET HARD DISK  (these are GPT partitions and legacy Windows will not install to a GPT disk).
5. Select blank target disk and install

If you legacy boot to Windows Setup it will only recognise and install to an MBR legacy partitioned disk.
If you UEFI boot to Windows Setup it will only recognise and install to a GPT partitioned disk.

This is standard Windows Setup behaviour and nothing to do with Ventoy.

**Comment by Gravarty** [comment]
Thanks the reply. The laptop does not support UEFI and is legacy only. The hard drive is set to MBR and was also reformed with diskpart (and was set to MBR with "convert mbr"), This bug never happened when booting Windows Setup with DVD or Rufus, only with Ventoy.

**Comment by steve6375** [comment]
Ventoy is now very well used by thousands of people. This problem is typical of a 'bad' target hard disk.
Clearly Windows Setup has a problem with the target drive.
So you need to give details.
Presumably you have done:
SHIFT+F10
DISKPART
LIST DISK
SEL DISK 0
CLEAN
CONVERT MBR
LIST DISK
EXIT

What system? What is target drive? Is it clean and MBR? Try a different target hard disk. Try running from a different system.

**Comment by Gravarty** [comment]
This is the second legacy system I have tried to install Windows with Ventoy and it has the same problem. The HDD is new and not damaged. It works without issues by booting Windows directly with a DVD or Rufus USB. I can format the drive with diskpart and see the new partition and browse the drive with cd/dir, but Windows Setup just refuses to install on this drive and shows the error message.

**Comment by nguyentumine** [comment]
@steve6375 In UEFI mode, if the boot disk is in GPT style then it should not contain an ESP partition. In Legacy mode it should not be hd0. I have tested on many systems and indeed many systems are affected by the above causes.

**Comment by steve6375** [comment]
Can you attach log files from Windows Setup?

**Comment by Gravarty** [comment]
Here's the log message:

`2023-05-05 04:30:19, Error      [0x06069d] IBS    GetMachineInfo:Couldn't find info for boot disk [1]
2023-05-05 04:31:39, Error      [0x06069d] IBS    GetMachineInfo:Couldn't find info for boot disk [1]
`

**Comment by steve6375** [comment]
What is the Ventoy USB drive?
Maybe try a previous version of Ventoy? A similar error was reported previously and was fixed in v1.0.80  #1777

**Comment by Jefffish09** [comment]
> @steve6375 In UEFI mode, if the boot disk is in GPT style then it should not contain an ESP partition. In Legacy mode it should not be hd0. I have tested on many systems and indeed many systems are affected by the above causes.
> 
> 
> 
> Even if you delete and create partitions manually, the installer will not accept installation on the partitions that you have created.

How to avoid the hd0? I have the same issue in Legacy mode, but I have only one SSD in my laptop.

**Comment by steve6375** [comment]
Have you set the boot disk in the bios settings to be the internal SSD?
You then boot to the Ventoy USB drive using an F key, e.g. F8 or F11 (depends on system)
Use bios boot selection  menu.

**Comment by Jefffish09** [comment]
Yes, the first boot device is the SSD. Do you mean that I should change the SSD to the second or third boot device?

> Have you set the boot disk in the bios settings to be the internal SSD?
> 
> You then boot to the Ventoy USB drive using an F key, e.g. F8 or F11 (depends on system)
> 
> Use bios boot selection  menu.
> 
>

**Comment by steve6375** [comment]
You are not the OP, so your setup (and experiences) may be different. Are you experienced in Legacy/UEFI and MBR\GPT partitions and Windows installs?
Boot device should be set as the internal SSD. Then use Fn to select USB boot device.
AFAIK no (or very few) people have reported this issue - and there should be thousands of people complaining if it affected them also.
So you will need to explain your particular issue fully. What system, what Ventoy USB, legacy or UEFI booting, did you CLEAN and CONVERT the target drive correctly before trying to install, did you delete ALL partitions from the target drive, what ISo are you using, etc.

**Comment by JH-IT** [comment]
I also have the same issue. 
ISO: latest Windows 10 22H2 x64 (just downloaded it via the Media Creation Tool).
Device: 10+years old (Lenovo T510), which doesn't support UEFI at all.
Ventoy Version: v1.0.94 (also tried v1.0.89 before)
SSD: completly empty, no partitions - cleaned via shredos.

When I try to boot the Windows setup in Ventoy, I can't boot in normal mode. When I try, all I see is a black screen with a blinking underscore in the top left corner. In wimboot it boots fine, but there I get the error from OP. I tried creating a partition and deleting it in the install setup and also converting to mbr via diskpart using the steps from @steve6375 https://github.com/ventoy/Ventoy/issues/2418#issuecomment-1573378855, but without luck.

I created a USB drive only with Windows via the Media Creation Tool and there I can boot just fine and it creates the partitions without problems. Note: When booting from that USB drive, I see the same black screen with the underscore, but I after a few seconds the installation screen appears. Via Ventoy I'm in an endless loop.

**Comment by shapitox** [comment]
As the topicstarter, recieved error "We couldn't create a new partition or locate an existing one..."
With diskpart I cleaned disk and do convert mbr - with no luck. It's strange, but when you click create partition button on clean unpartitioned disk the installer generally warns you that it needs to create system partitions, but in the case of ventoy it doesn't say anythink and creates only one primary partition on which you can't install the system with the above error. Ventoy 1.0.97 Intel Core E7500 4GB SSD120GB Legacy bios MBR disk. Tried a few msdn .iso win10/win11.

**Comment by KrisKnez** [comment]
I had an issue where I was receiving the error message
```
We couldn't create a new partition or locate an existing one. For more information, see the Setup log files
```

The log file in `C:/Windows/panther/setuperror.txt` contained
```
[0x06069d] IBS GetMachineInfo: Couldn't find info for boot disk [1]
```

The issue for me was that my BIOS was password locked (I lost the password) and the Windows installer wasn't able to update the boot order which apparently it does during the installation process.

The solution is to create a temporary installation partition on the disk and boot from it.

The steps are as follows

```
> diskpart
diskpart> list disk
diskpart> select disk 0   # Choose the disk you want to install to
diskpart> clean           # will delete everything, hope you have backups!
diskpart> create partition primary size=100000 # size in MB
diskpart> create partition primary size=7000 # temporary install partition
diskpart> format fs=ntfs quick
diskpart> assign          # assigns a drive letter, it was C: for me
diskpart> active          # makes the install partition bootable
diskpart> exit
```

Then copy the installation files to the newly temporary partition.
```
c:\> xcopy e: c: /e /h    # /e for recursive with empty, /h for system files
```
The command is copying all files from the `e:` to `c:`.
In my case `e:` was the virtual disk created by Ventoy that contains the installation files from the selected iso file while `c:` was the partition of the 100GB partition we created earlier for Windows.

Now just reboot without your USB and it will boot into the Windows installer from your disk. Proceed with the installation as normal.

After installation is complete Windows boot loader will give you the option between Windows and Windows Setup, choose Windows.

After Booting into Windows you can delete the Windows Setup partition because it is not needed anymore.

**Comment by EternalllZM** [comment]
Still an issue in 1.0.99, attempting to install Windows 10 IoT LTSC 2021 to a legacy BIOS device in wimboot mode results in the error
`We couldn't create a new partition or locate an existing one. For more information, see the Setup log files`

```
> diskpart
> list disk
> select disk 0
> clean
> convert mbr
> exit
```

same error

**Comment by heston88** [comment]
I can confirm this problem using a Thinkpad T500 with Windows 10 22H2v1 using an ssd with no partition table.

**Comment by omgtehlion** [comment]
I got the same error wile trying to install win10 x86. It did not boot in normal mode at all. And in wimboot mode I got this `We couldn't create a new partition or locate an existing one...`

Thanks, @KrisKnez, this method with xcopy to a temporary partition worked like a charm!

**Comment by Torkin1** [comment]
> ```
> c:\> xcopy e: c: /e /h    # /e for recursive with empty, /h for system files
> ```
> The command is copying all files from the `e:` to `c:`.
> In my case `e:` was the virtual disk created by Ventoy that contains the installation files from the selected iso file while `c:` was the partition of the 100GB partition we created earlier for Windows.

There is a typo I think. The C: volume should refer to the 7GB partition, not the 100GB one.

Apart from that, the rest of the solution works for me.

From what I have see, it seems like the windows setup incorrectly mistakes the usb drive as an available drive to install the OS, and queries the BIOS for informations about it.
